src/ape/api/providers.py

```diff
@@ -216,29 +274,59 @@ def await_confirmations(self) -> "ReceiptAPI":
 
 @abstractdataclass
 class BlockGasAPI:
+    """
+    An abstract class for representing gas data for a block.
+    """
+
     gas_limit: int
     gas_used: int
     base_fee: Optional[int] = None
 
     @classmethod
     @abstractmethod
     def decode(cls, data: Dict) -> "BlockGasAPI":
-        ...
+        """
+        Decode data to a :class:`~ape.api.BlockGasAPI`.
+
+        Args:
+            data (dict): A dictionary of block-gas properties.
+
+        Returns:
+            :class:`~ape.api.BlockGasAPI`
+        """
 
 
 @abstractdataclass
 class BlockConsensusAPI:
+    """
+    An abstract class representing block consensus-data,
+    such as PoW-related information regarding the block.
+    `EIP-3675` <https://eips.ethereum.org/EIPS/eip-3675>`__.
```

Comment:
```suggestion
    See `EIP-3675` <https://eips.ethereum.org/EIPS/eip-3675>`__ for an example of potential changes.
```

Reply:
We will have to open a separate PR for any suggestion related to providers.